File: facade/DBMS.java
package facade;

import java.util.HashMap;
import java.util.Locale;

public class DBMS {
    private HashMap<String, Row> db = new HashMap<String, Row>();

    public DBMS() {
        db.put("jane", new Row("Jane", "1990-01-01", "jane90@google.com"));
        db.put("robert", new Row("Robert", "1999-12-31", "robert99@google.com"));
        db.put("dorosh", new Row("Dorosh", "1988-12-01", "dorosh88@google.com"));
    }

    public Row query(String name) {
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return db.get(name.toLowerCase());
    }
}

File: ts/src/facade/DBMS.ts
import Row from "./Row";

export default class DBMS {
  private db = new Map<string, Row>();

  constructor() {
    this.db.set("jane", new Row("jane", "1990-02-14", "jane09@geosee.co.kr"));
    this.db.set(
      "robert",
      new Row("robert", "1994-03-12", "robert@geosee.co.kr")
    );
    this.db.set(
      "dorosh",
      new Row("dorosh", "1999-06-12", "dorosh@geosee.co.kr")
    );
  }

  qeury(name: string, callback: (row: Row) => void): void {
    setTimeout(() => {
      callback(this.db.get(name));
    }, 1500);
  }
}
